src/Umbraco.Core/Persistence/SqlSyntax/SqlServerSyntaxProvider.cs

```diff
@@ -286,15 +306,24 @@ public override void ReadLock(IDatabase db, params int[] lockIds)
             if (db.Transaction.IsolationLevel < IsolationLevel.ReadCommitted)
                 throw new InvalidOperationException("A transaction with minimum ReadCommitted isolation level is required.");
 
-            // *not* using a unique 'WHERE IN' query here because the *order* of lockIds is important to avoid deadlocks
+            var timeout = TimeSpan.FromMilliseconds(Current.Configs.Global().SqlWriteLockTimeOut);
+
             foreach (var lockId in lockIds)
             {
-                var i = db.ExecuteScalar<int?>("SELECT value FROM umbracoLock WITH (REPEATABLEREAD) WHERE id=@id", new { id = lockId });
-                if (i == null) // ensure we are actually locking!
-                    throw new ArgumentException($"LockObject with id={lockId} does not exist.", nameof(lockIds));
+                ObtainReadLock(db, timeout, lockId);
             }
         }
 
+        private static void ObtainReadLock(IDatabase db, TimeSpan timeout, int lockId)
+        {
+            db.Execute(@"SET LOCK_TIMEOUT " + timeout.TotalMilliseconds  + ";");
```

Comment:
This is new, this was never done before when obtaining a readlock, see https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Core/Persistence/SqlSyntax/SqlServerSyntaxProvider.cs#L282